src/poc/miner_onion_server.erl

```diff
@@ -232,10 +240,9 @@ wait_for_block_(Fun, Count) ->
             wait_for_block_(Fun, Count - 1)
     end.
 
-decrypt(Type, IV, OnionCompactKey, Tag, CipherText, RSSI, Stream, #state{ecdh_fun=ECDHFun}=State) ->
+decrypt(Type, IV, OnionCompactKey, Tag, CipherText, RSSI, Stream, #state{ecdh_fun=ECDHFun, chain = Chain}=State) ->
```

Comment:
What if chain is undefined here?

Reply:
if chain in undefined, we will never hit decrypt...all casts whilst chain is undefined will be caught by the new handler